cpp/include/cudf_test/tdigest_utilities.cuh

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
```

Comment:
Is cudf_test the right place for this? I generally see such headers that pertain to only one functionality in the same dir as the test source.

Reply:
This code is used by both the reduction and groupby tests (for the cases where the groupby is doing simple tests with 1 key, they end up the same)